### ISSUE TITLE
[ktx2] Running toktx with no arguments triggers a segmentation fault

### ISSUE THREAD

**Issue by zeux** [KhronosGroup/KTX-Software#143]
The steps to repro are just to build a release version of toktx and run it without arguments. This looks like a std::string construction from a NULL pointer.

**Comment by MarkCallow** [comment]
I am having a hard time reproducing this. It has failed once. I then set a breakpoint in `malloc_error_break` and it didn't fail. I removed the breakpoint. It didn't fai.l I restarted Xcode. It didn't fail. I tried many times. It didn't fail. This smells like an uninitialized variable. If you can more reliably reproduce it, please see if you can narrow down the cause.

**Comment by zeux** [comment]
The callstack I get in Debug is as follows:

```
(gdb) bt
#0  __memmove_avx_unaligned_erms () at ../sysdeps/x86_64/multiarch/memmove-vec-unaligned-erms.S:238
#1  0x00007fffff508ae5 in std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::_M_assign(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) () from /usr/lib/x86_64-linux-gnu/libstdc++.so.6
#2  0x0000000008015af9 in std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::assign (__str=...,
    this=0x7ffffffee0d8) at /usr/include/c++/9/bits/basic_string.h:1364
#3  std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::operator= (__str=..., this=0x7ffffffee0d8)
    at /usr/include/c++/9/bits/basic_string.h:696
#4  processCommandLine (argc=1, argv=0x7ffffffee458, options=...) at /mnt/c/work/KTX-Software/tools/toktx/toktx.cpp:1169
#5  0x0000000008016139 in main (argc=1, argv=0x7ffffffee458) at /mnt/c/work/KTX-Software/tools/toktx/toktx.cpp:590
```

So it crashes on this line:
```
    options.outfile = parser.argv[i++];
```

Here, `parser.optind` is 1, so `i` starts as 1; `argc` is also 1 so this is dereferencing the element past the end of `argv` array - on my system it's always NULL, maybe on macOS it's sometimes non-NULL.

**Comment by MarkCallow** [comment]
Fixed by PR #147. Autoclose did not work despite the PR saying fixes #143.
